apis/duck/v1/podspec_types.go

```diff
@@ -82,6 +84,11 @@ func (*WithPod) GetListType() runtime.Object {
 	return &WithPodList{}
 }
 
+// GetGroupVersionKind returns a GroupVersionKind.
+func (*WithPod) GetGroupVersionKind() schema.GroupVersionKind {
```

Comment:
Not sure if this is the right thing to do here as `WithPod` doesn't really exist. Should this read the actual GroupVersionKind from the typeMeta?